dashboard/lib/services/lti.rb

```diff
@@ -42,4 +42,107 @@ def self.get_claim_from_list(id_token, keys_array)
   def self.get_claim(id_token, key)
     id_token[key] || id_token.dig(Policies::Lti::LTI_CUSTOM_CLAIMS, key)
   end
+
+  def self.initialize_lti_student_from_nrps(client_id:, issuer:, nrps_member:)
+    # TODO DAYNE don't do [0]
+    custom_claims = nrps_member[:message][0][Policies::Lti::LTI_CUSTOM_CLAIMS.to_sym]
+    user = User.new
+    user.provider = User::PROVIDER_MIGRATED
+    user.user_type = User::TYPE_STUDENT
+    user.name = custom_claims[:display_name] || custom_claims[:full_name] || custom_claims[:given_name]
```

Comment:
Schoology actually has name values at the root level of the id_token. We might want to use this helper function that I wrote to dig the value out from either root or custom_claims https://github.com/code-dot-org/code-dot-org/blob/staging-next/dashboard/lib/services/lti.rb#L38

Reply:
Implemented in `initialize_lti_user`.

Reply:
Yeah it's in a different spot, but the key is the same - so I was able to just pass in the object from the response that has that key.